src/dpl/src/OptMirror.cpp

```diff
@@ -143,28 +159,177 @@ Opendp::findMirrorCandidates(NetBoxes &net_boxes,
+  map<dbNet*, vector<dbITerm *>> net2iterm;
+
+  for (dbITerm *iterm : inst->getITerms()) {
+    if (iterm->getNet() != NULL) {
+      dbNet *net = iterm->getNet();
+
+      if (net2iterm.find(net) == net2iterm.end()) {
+        net2iterm[net] = vector<dbITerm *>();
+        net2iterm[net].push_back(iterm);
+      } else {
+        net2iterm[net].push_back(iterm);
+      }
+    }
+  }
+
+  for ( auto &item : net2iterm ) {
+    dbNet *net = item.first;
+    if (!isSupply(net)) {
+      Rect *new_net_box = new Rect();
+      delta += hpwlIncrementHelper(inst, item.second, net, pt_x, mirror,
+                                  new_net_box, net2box[net]);
+      updated_box.push_back({net, new_net_box});
+    }
+  }
+  return delta;
+}
+
+int64_t
+Opendp::hpwlIncrementHelper(dbInst *inst, vector<dbITerm *> &iterms,
+                      dbNet *net, int pt_x, bool mirror, Rect *new_net_box,
+                      Rect netBox) {
+  int64_t net_hpwl = netBox.dx() + netBox.dy();
+  Rect iterm_box;
+  iterm_box.mergeInit();
+  int cellPtX, cellPtY;
+  inst->getLocation(cellPtX, cellPtY);
+  int cellWidth = inst->getMaster()->getWidth();
+
+  for (int i = 0; i < iterms.size(); i++) {
+    dbITerm *iterm = iterms[i];
+    int x, y;
+    if (iterm->getAvgXY(&x, &y)) {
+      Rect iterm_rect(x, y, x, y);
+      iterm_box.merge(iterm_rect);
+    } else {
+      // This clause is sort of worthless because getAvgXY prints
+      // a warning when it fails.
+      dbInst *inst = iterm->getInst();
+      odb::dbBox *inst_box = inst->getBBox();
+      int center_x = (inst_box->xMin() + inst_box->xMax()) / 2;
+      int center_y = (inst_box->yMin() + inst_box->yMax()) / 2;
+      Rect iterm_rect(center_x, center_y, center_x, center_y);
+      iterm_box.merge(iterm_rect);
+    }
+  }
+
+  // Based on current location check the pin location in the netbox
+  bool isInside = netBox.inside(iterm_box);
+
+  // Calculate the new pin delta displacement
+  int mirror_x = 0;
+  if (mirror) {
+    mirror_x = 2 * cellPtX + cellWidth - iterm_box.xMin() - iterm_box.xMax();
+  }
+
+  int delta_x = pt_x - cellPtX + mirror_x;
+
+  // Considering there wont be any movement along the y axis
+  iterm_box.moveDelta(delta_x, 0);
+  bool isContain = netBox.contains(iterm_box);
+
+  // Pin term is inside the net box before and after movement.
+  if (isInside && isContain) {
+    new_net_box->reset(netBox.xMin(), netBox.yMin(), netBox.xMax(),
+                        netBox.yMax());
+    return 0;
+  }
+
+  // Pin term is initially inside the net box but after
+  // movement is outside the netbox
+  if (isInside) {
+    netBox.merge(iterm_box);
+    int64_t new_hpwl = netBox.dx() + netBox.dy();
+    new_net_box->reset(netBox.xMin(), netBox.yMin(), netBox.xMax(),
+                        netBox.yMax());
+    return new_hpwl - net_hpwl;
+  }
+
+  // Re calculate the Net box with the updated iterm location
+  new_net_box->mergeInit();
+
+  for (dbITerm *iterm_ : net->getITerms()) {
```

Comment:
variables ending in underscore should be class variables, not local variables.

Reply:
I have fixed this in the latest commit.